homeassistant/components/airthings/config_flow.py

```diff
@@ -0,0 +1,63 @@
+"""Config flow for Airthings integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import airthings
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_ID, CONF_SECRET, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ID): str,
+        vol.Required(CONF_SECRET): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Airthings."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            await airthings.get_token(
+                async_get_clientsession(self.hass),
+                user_input[CONF_ID],
+                user_input[CONF_SECRET],
```

Comment:
Why do we not use the oauth2 code flow instead of the client credentials flow?

https://developer.airthings.com/docs/api-authorization

Reply:
Yes, that is only for business accounts.
Here is the consumer API: https://developer.airthings.com/consumer-api-docs/